### ISSUE TITLE
EXP1HCL brake PWM is not applied under some conditions

### ISSUE THREAD

**Issue by dc42** [dc42/RepRapFirmware#1050]
If I run the following sequence:
1. Run a macro that includes the following lines (note that "brake" is not a valid pin number):
```
M584 X123.0
M569.7 P123.0 C"brake" V24
M569.1 P123.0 T2 S200 C1000 R15 I200 D0.02 A250000 V800
M906 X3000
M569 P123.0 D2
M17 X
```

2. Run the same macro with "brake" corrected to "brake.neg"

then brake PWM is no longer applied, the full voltage is applied to the brake. This bug is present in at least 3.5.3 and in 3.6.0-beta.1.

**Comment by dc42** [comment]
Simpler way to reproduce this issue: enable the motor using M17 before using M569.7 to configure the brake port.

**Comment by dc42** [comment]
Fixed and tested in 3.5-dev and 3.6-dev.
